### PR TITLE
Update the demo of paddle.grad

### PR DESCRIPTION
**Pull request by wzzju**
### PR types

Others
### PR changes

Docs

### Describe

Update the demo code of `paddle.grad`. [The pr of FluidDoc about paddle.grad](https://github.com/PaddlePaddle/FluidDoc/pull/2444).
image
image
image

### CODE REVIEW

**Review by willthefrog — COMMENTED** [review]
maybe simplify the examples a bit, something like this should suffice
```python

x = paddle.ones(shape=[1], dtype='float32')
x.stop_gradient = False
y = x * x

dx = paddle.grad(
        outputs=[y],
        inputs=[x],
        create_graph=create_graph,
        retain_graph=True)[0]

print(dx)
# value should be ....
```

**Comment by wzzju** [comment]
> maybe simplify the examples a bit, something like this should suffice
> 
> ```python
> x = paddle.ones(shape=[1], dtype='float32')
> x.stop_gradient = False
> y = x * x
> 
> dx = paddle.grad(
>         outputs=[y],
>         inputs=[x],
>         create_graph=create_graph,
>         retain_graph=True)[0]
> 
> print(dx)
> # value should be ....
> ```

Thanks for your advice. These example codes here are usefully, better left unchanged.

File: python/paddle/fluid/dygraph/base.py [hunk 1]
```diff
@@ -375,47 +375,46 @@ def grad(outputs,
...
+                # x.gradient() = 2 * x + 2 = 4.0
+
+                z.backward()
+                return x.gradient()
+
+            print(test_dygraph_grad(create_graph=False)) # [2.]
             print(test_dygraph_grad(create_graph=True)) # [4.]
 
     Examples 2:
         .. code-block:: python
```
Comment (jzhang533) [inline python/paddle/fluid/dygraph/base.py:412]: exapmle2 里不需要再定义一个函数了吧。

File: python/paddle/fluid/dygraph/base.py [hunk 2]
```diff
@@ -431,15 +430,15 @@ def test_dygraph_grad(grad_outputs=None):
...
                     outputs=[y1, y2], 
                     inputs=[x],
                     grad_outputs=grad_outputs)[0]
 
                 return dx.numpy()
 
-            THREE = fluid.layers.fill_constant(shape=[1], value=3.0, dtype='float32')
-            FOUR = fluid.layers.fill_constant(shape=[1], value=4.0, dtype='float32')
+            THREE = paddle.fill_constant(shape=[1], value=3.0, dtype='float32')
+            FOUR = paddle.fill_constant(shape=[1], value=4.0, dtype='float32')
```
Comment (jzhang533) [inline python/paddle/fluid/dygraph/base.py:441]: THREE, FOUR是干嘛的？

File: python/paddle/fluid/dygraph/base.py (hunk 1 above)
Comment (wzzju) [reply python/paddle/fluid/dygraph/base.py:412]: 感谢。为了展示参数不同时不同的调用效果（代码复用），所以在example代码中定义了函数。

File: python/paddle/fluid/dygraph/base.py (hunk 2 above)
Comment (wzzju) [reply python/paddle/fluid/dygraph/base.py:441]: Have changed the var name. Thanks.

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by willthefrog — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]

**Pull request merged**